src/connector/src/sink/encoder/json.rs

```diff
@@ -85,28 +95,34 @@ impl JsonEncoder {
         col_indices: Option<Vec<usize>>,
         map: HashMap<String, u8>,
     ) -> Self {
-        Self {
-            schema,
-            col_indices,
+        let config = JsonEncoderConfig {
             time_handling_mode: TimeHandlingMode::Milli,
             date_handling_mode: DateHandlingMode::String,
             timestamp_handling_mode: TimestampHandlingMode::String,
             timestamptz_handling_mode: TimestamptzHandlingMode::UtcWithoutSuffix,
             custom_json_type: CustomJsonType::Doris(map),
+        };
+        Self {
+            schema,
+            col_indices,
             kafka_connect: None,
+            config,
         }
     }
 
     pub fn new_with_starrocks(schema: Schema, col_indices: Option<Vec<usize>>) -> Self {
-        Self {
-            schema,
-            col_indices,
+        let config = JsonEncoderConfig {
```

Comment:
nit: These can be `new_json_encoder_config` in each individual sinks (`sink/starrocks.rs`). But we can leave it for later.